src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -27,21 +27,21 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             StorageCredentials credentials = account.Credentials;
 
             // Avoid double-validating the same account and credentials.
-            if (_validatedCredentials.Contains(credentials))
+            if ((!isPrimaryAccount && _validatedCredentials.Contains(credentials)) || _primaryCredentials == credentials)
```

Comment:
It's a corner case if a secondary account and a primary account use the same connection string.

1. Blob only storage account as secondary storage is validated (does not do queue check)
2. The same account is also used as primary
3. Because it already exists in validatedCredentials, the queue check is skipped

Might not be worth the more complex code, I doubt this will get hit and if it does we'll throw an error anyway (though likely more cryptic)